admin/src/containers/MainLayout/PanelContext.js

```diff
@@ -0,0 +1,53 @@
+import React, { Component, createContext } from 'react';
+import PropTypes from 'prop-types';
+
+const PanelContext = createContext({
+  component: null,
+  title: null,
+  props: {},
+  openPanel: () => {},
+  closePanel: () => {},
+  isPanelOpen: false
+});
+
+export class PanelProvider extends Component {
+  static propTypes = {
+    children: PropTypes.oneOfType([
+      PropTypes.element,
+      PropTypes.object,
+      PropTypes.arrayOf(PropTypes.element)
+    ])
+  };
+
+  openPanel = (component, title = null, props = {}) => {
```

Comment:
Why not just `openPanel() {}` ?

Reply:
I don't understand the question? The `PanelProvider` needs to provide an `openPanel` function to the components that want to interact with this context. I'm setting the state so that the component gets re-rendered. That is how we can simulate something similar to actions and reducers that change the central store similarly to redux. Let's discuss this today and I'll refactor in a separate task if you have something better in mind.